Program.cs

```diff
@@ -1,10 +1,85 @@
-﻿namespace Lekce_12_HW
+﻿using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Lekce_12_HW
 {
     internal class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            Console.WriteLine("Data z webové služby:");
+            await GetPublicApiDataAsync();
         }
+
+        private static async Task GetPublicApiDataAsync()
+        {
+            try
+            {
+                using HttpClient client = new HttpClient();
+                const string apiKlic = "";
+                const string mesto = "Prague";
+                string url = $"https://api.weatherapi.com/v1/current.json?key={apiKlic}&q={mesto}&aqi=no";
+
+                HttpResponseMessage response = await client.GetAsync(url);
+                response.EnsureSuccessStatusCode();
+
+                string responseBody = await response.Content.ReadAsStringAsync();
+                Console.WriteLine("Odpověď z API:");
+                Console.WriteLine(responseBody);
+                DeserializacePocasi(responseBody);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Nastala chyba: ");
+            }
+        }
+
+        private static void DeserializacePocasi(string jsonData)
+        {
+            try
+            {
+                var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true }; // bez tohodle mi to proste nechtelo fungovat pomohlo az ChatGPT :/
+                var weatherData = JsonSerializer.Deserialize<WeatherDTO>(jsonData, options);
+
+                if (weatherData != null)
+                {
+                    Console.WriteLine($"Počasí v {weatherData.Location.Name}:");
+                    Console.WriteLine($"Teplota: {weatherData.Current.TempC}°C");
+                    Console.WriteLine($"Stav: {weatherData.Current.Condition.Text}");
+                }
+                else
+                {
+                    Console.WriteLine("Nepodařilo se deserializovat data.");
+                }
+            }
+            catch (JsonException ex)
+            {
+                Console.WriteLine($"Chyba při deserializaci JSON: {ex.Message}");
+            }
+        }
+    }
+
+    public class WeatherDTO
+    {
+        public Location Location { get; set; }
+        public Current Current { get; set; }
+    }
+
+    public class Location
+    {
+        public string Name { get; set; }
+    }
+
+    public class Current
+    {
+        public double TempC { get; set; }
+        public Condition Condition { get; set; }
+    }
+
+    public class Condition
+    {
+        public string Text { get; set; }
```

Comment:
Protoze vetsinou to nematchuje 1:1, tak se pomoci atributu doplnuje informace o tom, jak se to ma (at uz pro deserializaci, nebo serializaci) jmenovat.

Takto by to vypadalo napsane spravne:
```suggestion
    public class WeatherDTO
    {
        [JsonPropertyName("location")]
        public Location Location { get; set; }
        [JsonPropertyName("current")]
        public Current Current { get; set; }
    }

    public class Location
    {
        [JsonPropertyName("name")]
        public string Name { get; set; }
    }

    public class Current
    {
        [JsonPropertyName("temp_c")]
        public double TempC { get; set; }
        [JsonPropertyName("condition")]
        public Condition Condition { get; set; }
    }

    public class Condition
    {
        [JsonPropertyName("text")]
        public string Text { get; set; }
```

Reply:
ta `PropertyNameCaseInsensitive` option funguje tak, ze rekne deserializeru, ze chces matchovat property necitlive na jejich casing (velka a mala pismena). Nevim ted z hlavy, jaktoze ti to matchuje i to temp_c, kdyz tam je navic ten `_` znak, ale takhle vecer jsem uz nemel silu to zjistovat, promin 🙈 ale snad ted uz dava smysl, proc to reseni, co ti poradil AI bot funguje a proc to neni z obecneho hlediska idealni. Lepsi je explicitne vedet, co na co mapujes a definovat ta jmena takhle natvrdo.

Reply:
myslim si (odhad), ze ten temp_c funguje, protoze to je jeden z druhu zapisu (snake case), takze ten deserializer nejspis bere temp_c, TempC, i treba temp-c jako stejne hodnoty